src/js_native_api_v8.cc

```diff
@@ -38,6 +38,17 @@
 #define CHECK_NEW_FROM_UTF8(env, result, str)                                  \
   CHECK_NEW_FROM_UTF8_LEN((env), (result), (str), NAPI_AUTO_LENGTH)
 
+#define CHECK_NEW_STRING_ARGS(env, str, length, result)                        \
```

Comment:
I realize you're riffing off an existing pattern here but:

1. It's considered good style to name control flow macros in a way that makes it obvious what they do (e.g. RETURN_IF_FOO)

2. All the more so because CHECK_FOO is a widely used pattern inside node that means something completely different

Maybe something for a follow-up PR.

Reply:
I agree this can be done in a follow-up since this is the pattern in the codebase.